tests/sys/psa_crypto_cipher/example_cipher_aes_128.c

```diff
@@ -81,5 +81,7 @@ psa_status_t example_cipher_aes_128(void)
     if (status == PSA_SUCCESS) {
         return (memcmp(PLAINTEXT, plain_out, sizeof(plain_out)) ? -1 : 0);
     }
+
+    status = psa_destroy_key(key_id);
```

Comment:
Just had a look at you `psa_destroy_key` problem:

```suggestion
    if (status != PSA_SUCCESS) {
        return status;
    }

    status = psa_destroy_key(key_id);
    if (status != PSA_SUCCESS) {
        return status;
    }

    return memcmp(PLAINTEXT, plain_out, sizeof(plain_out)) ? PSA_ERROR_INVALID_SIGNATURE : PSA_SUCCESS;
```

and same below. With that, decreasing the key count back to 1 should work :)